### PR TITLE
[API] Refactor StateProviders

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/StateProvider/Admin/Account/ResetPassword/ItemProvider.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider\Admin\Account\ResetPassword;
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/StateProvider/Admin/Account/ResetPassword/ItemProvider.php]:
```suggestion
namespace Sylius\Bundle\ApiBundle\StateProvider\Admin\ResetPassword;
```
or
```suggestion
namespace Sylius\Bundle\ApiBundle\StateProvider\Admin\AdminUser\ResetPassword;
```
but as the command is in the `Account` directory, the current namespace has probably more sense 🤔

File: src/Sylius/Bundle/ApiBundle/StateProvider/Shop/Account/VerifyShopUser/ItemProvider.php [hunk 2]
```diff
@@ -11,29 +11,31 @@
 
 declare(strict_types=1);
 
-namespace Sylius\Bundle\ApiBundle\StateProvider\Input;
+namespace Sylius\Bundle\ApiBundle\StateProvider\Shop\Account\VerifyShopUser;
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/StateProvider/Shop/Account/VerifyShopUser/ItemProvider.php]: Similar as with the admin side:
```suggestion
namespace Sylius\Bundle\ApiBundle\StateProvider\Shop\VerifyShopUser;
```
or
```suggestion
namespace Sylius\Bundle\ApiBundle\StateProvider\Shop\ShopUser\VerifyShopUser;
```
but I'm not 100% convinced 🤔

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/StateProvider/Admin/Account/ResetPassword/ItemProvider.php (hunk 1 above)
Comment (mpysiak) [reply src/Sylius/Bundle/ApiBundle/StateProvider/Admin/Account/ResetPassword/ItemProvider.php]: 👍 in favour of keeping namespace as flat as possible:

```
namespace Sylius\Bundle\ApiBundle\StateProvider\Admin\ResetPassword;
```

File: src/Sylius/Bundle/ApiBundle/StateProvider/Shop/Account/VerifyShopUser/ItemProvider.php (hunk 2 above)
Comment (mpysiak) [reply src/Sylius/Bundle/ApiBundle/StateProvider/Shop/Account/VerifyShopUser/ItemProvider.php]: Same here

**Review by mpysiak — APPROVED** [review]

**Review by Rafikooo — APPROVED** [review]

**Pull request merged**